src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -408,7 +465,7 @@ protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceTe
                         return textContainer.TryGetTextBuffer();
                     });
 
-                    UpdateText(textBuffer, text);
+                    UpdateText(textBuffer!, text);
```

Comment:
This seems fishy, do if (textBuffer != null) surrounding it?

(FWIW, this exact type of bug is what made us have to re-ship 16.4 Preview 3, so I have learned my lesson. :smile:)

Reply:
Can do that, but the ! doesn't really change behavior - it would null ptr in UpdateText before anyway - http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices.LiveShare/Client/RemoteLanguageServiceWorkspace.cs,426

Reply:
Yeah this isn't new, but it's still bad. :-/ Yes, it doesn't strictly need to be fixed in *this* PR, but I'd be scared otherwise.